macOS/DuckDuckGo/Subscription/AuthV2PixelHandler.swift

```diff
@@ -43,8 +46,34 @@ public struct AuthV2PixelHandler: SubscriptionPixelHandler {
     }
 
     let source: Source
+    private let notificationCenter: NotificationCenter = NotificationCenter.default
+    private var cancellables = Set<AnyCancellable>()
 
-    public func handle(pixelType: Subscription.SubscriptionPixelType) {
+    init(source: Source) {
+        self.source = source
+
+        notificationCenter.publisher(for: .subscriptionDidChange).sink { param in
+
+            guard let userInfo = param.userInfo as? [AnyHashable: PrivacyProSubscription],
+                  let subscription = userInfo[UserDefaultsCacheKey.subscription] else {
+                return
+            }
+
+            if !subscription.isActive {
+                PixelKit.fire(PrivacyProPixel.privacyProSubscriptionExpired(source), frequency: .daily)
+            }
+        }.store(in: &cancellables)
+
+        notificationCenter.publisher(for: .entitlementsDidChange).sink { param in
```

Comment:
There is a problem with this notification in V2 as it is also posted in `DefaultSubscriptionEndpointServiceV2` where the value of entitlements are subscription features. So this notification can be posted by an expired user (with no entitlements) where the array will contain all values.

Reply:
I've addressed this point filtering the notification from the source, thanks.